File: August 2026/2026-08-05/19-remove-nth-node-from-end-of-list.cpp
// 19 leeetcode remove-nth-node-from-end-of-list.cpp;ll-2p;help

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        struct ListNode* dummy =  (struct ListNode*)malloc(sizeof(struct ListNode));
        dummy->next = head;
        struct ListNode* first = dummy;
        struct ListNode* second = dummy;
        for(int i = 0; i <= n; i++) {
            first = first->next;
        }
        while(first != NULL) {
            first = first->next;
            second = second->next;
        }

        struct ListNode* toDelete = second->next;
        second->next = second->next->next;
        struct ListNode* newHead = dummy->next;
        return newHead;
    }
};

File: September 2026/2026-09-17/19-remove-nth-node-from-end-of-list.py
# 19-remove-nth-node-from-end-of-list.py;ll;help;leetcode

# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: ListNode | None, n: int) -> ListNode | None:
        dummy = ListNode(0)
        dummy.next = head

        first = dummy
        second = dummy

        for _ in range(n + 1):
            first = first.next

        while first is not None:
            first = first.next
            second = second.next
        
        toDelete = second.next
        second.next = second.next.next

        return dummy.next